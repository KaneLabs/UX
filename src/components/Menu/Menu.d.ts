export default Menu;
declare class Menu extends React.Component<any, any, any> {
  static propTypes: {
    button: PropTypes.Requireable<PropTypes.ReactNodeLike>;
    children: PropTypes.Requireable<PropTypes.ReactNodeLike>;
    visible: PropTypes.Requireable<boolean>;
    menuStyle: PropTypes.Requireable<object>;
    sameWidth: PropTypes.Requireable<boolean>;
    onBackdropPress: PropTypes.Requireable<(...args: any[]) => any>;
    modalMenuStyle: PropTypes.Requireable<object>;
    contentContainerStyle: PropTypes.Requireable<object>;
    noBackDrop: PropTypes.Requireable<boolean>;
    tooltip: PropTypes.Requireable<boolean>;
    tooltipPosition: PropTypes.Requireable<string>;
    testID: PropTypes.Requireable<string>;
    fullWidth: PropTypes.Requireable<boolean>;
  };
  constructor(props: any);
  constructor(props: any, context: any);
  onModalMenuLayout: (e: any) => void;
  onButtonLayout: (e: any) => void;
  onMenuLayout: (e: any) => void;
  toggle(): void;
  marker: View | null | undefined;
}
import React from 'react';
import { View } from 'react-native';
import PropTypes from 'prop-types';
