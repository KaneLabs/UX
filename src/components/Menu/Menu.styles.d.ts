export default styles;
declare namespace styles {
  namespace menuContainer {
    const top: number;
    const overflow: 'hidden';
    const position: 'absolute';
    const zIndex: number;
  }
  const menu:
    | {
        shadowColor: string;
        shadowOffset: {
          width: number;
          height: number;
        };
        shadowOpacity: number;
        shadowRadius: number;
        borderRadius: number;
        position: 'absolute';
        left: number;
        top: number;
        backgroundColor: string;
        minWidth: number;
        zIndex: number;
      }
    | {
        elevation: any;
        borderRadius: number;
        position: 'absolute';
        left: number;
        top: number;
        backgroundColor: string;
        minWidth: number;
        zIndex: number;
      };
}
