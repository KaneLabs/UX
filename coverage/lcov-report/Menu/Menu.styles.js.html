
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for Menu/Menu.styles.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../index.html">All files</a> / <a href="index.html">Menu</a> Menu.styles.js</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>9/9</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>0/0</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>0/0</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>1/1</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line high'></div>
    <pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a></td><td class="line-coverage quiet"><span class="cline-any cline-yes">1x</span></td><td class="text"><pre class="prettyprint lang-js">var _interopRequireDefault=require("@babel/runtime/helpers/interopRequireDefault");Object.defineProperty(exports,"__esModule",{value:true});exports.default=void 0;var _extends2=_interopRequireDefault(require("@babel/runtime/helpers/extends"));var _reactNative=require("react-native");var _Shadow=_interopRequireDefault(require("../Shadow"));var styles=_reactNative.StyleSheet.create({menuContainer:{top:-10,overflow:'hidden',position:'absolute',zIndex:200},menu:(0,_extends2.default)({borderRadius:6,position:'absolute',left:10,top:10,backgroundColor:'white',minWidth:112,zIndex:200},(0,_Shadow.default)(8))});var _default=styles;exports.default=_default;</pre></td></tr></table></pre>

                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Wed Mar 31 2021 17:15:17 GMT-0600 (Mountain Daylight Time)
            </div>
        </div>
        <script src="../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../sorter.js"></script>
        <script src="../block-navigation.js"></script>
    </body>
</html>
    