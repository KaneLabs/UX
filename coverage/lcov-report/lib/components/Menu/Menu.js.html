
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for lib/components/Menu/Menu.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../index.html">All files</a> / <a href="index.html">lib/components/Menu</a> Menu.js</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">54.42% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>80/147</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">26.92% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>14/52</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">37.5% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>6/16</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>1/1</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line medium'></div>
    <pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a></td><td class="line-coverage quiet"><span class="cline-any cline-yes">3x</span></td><td class="text"><pre class="prettyprint lang-js">var _interopRequireDefault=require("@babel/runtime/helpers/interopRequireDefault");var _interopRequireWildcard=require("@babel/runtime/helpers/interopRequireWildcard");Object.defineProperty(exports,"__esModule",{value:true});exports.default=void 0;var _classCallCheck2=_interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));var _createClass2=_interopRequireDefault(require("@babel/runtime/helpers/createClass"));var _inherits2=_interopRequireDefault(require("@babel/runtime/helpers/inherits"));var _possibleConstructorReturn2=_interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));var _getPrototypeOf2=_interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));var _react=_interopRequireWildcard(require("react"));var _propTypes=_interopRequireDefault(require("prop-types"));var _Animated=_interopRequireDefault(require("react-native-web/dist/exports/Animated"));var _View=_interopRequireDefault(require("react-native-web/dist/exports/View"));var _Easing=_interopRequireDefault(require("react-native-web/dist/exports/Easing"));var _Platform=_interopRequireDefault(require("react-native-web/dist/exports/Platform"));var _ScrollView=_interopRequireDefault(require("react-native-web/dist/exports/ScrollView"));var _Menu=_interopRequireDefault(require("./Menu.styles"));var _ModalMenu=_interopRequireDefault(require("./ModalMenu"));var _jsxFileName="/Users/kane/Lab/Eros/src/components/Menu/Menu.js";function _createSuper(Derived){var hasNativeReflectConstruct=_isNativeReflectConstruct();return function _createSuperInternal(){var Super=(0,_getPrototypeOf2.default)(Derived),result;<span class="missing-if-branch" title="else path not taken" >E</span>if(hasNativeReflectConstruct){var NewTarget=(0,_getPrototypeOf2.default)(this).constructor;result=Reflect.construct(Super,arguments,NewTarget);}else{<span class="cstat-no" title="statement not covered" >result=Super.apply(this,arguments);}</span>return(0,_possibleConstructorReturn2.default)(this,result);};}function _isNativeReflectConstruct(){<span class="missing-if-branch" title="if path not taken" >I</span>if(typeof Reflect==="undefined"||!Reflect.construct)<span class="cstat-no" title="statement not covered" >return false;<span class="missing-if-branch" title="if path not taken" >I</span>i</span>f(Reflect.construct.sham)<span class="cstat-no" title="statement not covered" >return false;<span class="missing-if-branch" title="else path not taken" >E</span>i</span>f(typeof Proxy==="function")return true;<span class="cstat-no" title="statement not covered" >try{<span class="cstat-no" title="statement not covered" >Boolean.prototype.valueOf.call(Reflect.construct(Boolean,[],<span class="fstat-no" title="function not covered" >fu</span>nction(){}));<span class="cstat-no" title="statement not covered" >r</span>eturn true;}</span>catch(e){<span class="cstat-no" title="statement not covered" >return false;}</span>}</span>var Menu=function(_Component){(0,_inherits2.default)(Menu,_Component);var _super=_createSuper(Menu);function Menu(){var _this;(0,_classCallCheck2.default)(this,Menu);for(var _len=arguments.length,args=new Array(_len),_key=0;_key&lt;_len;_key++){args[_key]=arguments[_key];}_this=_super.call.apply(_super,[this].concat(args));_this.state={menuHeight:new _Animated.default.Value(0),menuWidth:new _Animated.default.Value(0),opacity:new _Animated.default.Value(0),modalMenuWidth:0,easing:_Easing.default.bezier(0.4,0,0.2,1),animationDuration:300,expanded:false};_this.onModalMenuLayout=<span class="fstat-no" title="function not covered" >fu</span>nction(e){var width=<span class="cstat-no" title="statement not covered" >e.nativeEvent.layout.width;<span class="cstat-no" title="statement not covered" ></span>_this.setState({modalMenuWidth:width});}</span>;_this.onButtonLayout=<span class="fstat-no" title="function not covered" >fu</span>nction(e){var _e$nativeEvent$layout=<span class="cstat-no" title="statement not covered" >e.nativeEvent.layout,</span>width=<span class="cstat-no" title="statement not covered" >_e$nativeEvent$layout.width,</span>height=<span class="cstat-no" title="statement not covered" >_e$nativeEvent$layout.height;</span>var locationX;var locationY;<span class="cstat-no" title="statement not covered" >if(_Platform.default.OS==='web'){<span class="cstat-no" title="statement not covered" >locationX=e.nativeEvent.target.getBoundingClientRect().x;<span class="cstat-no" title="statement not covered" >l</span>ocationY=e.nativeEvent.target.getBoundingClientRect().y;<span class="cstat-no" title="statement not covered" >_</span>this.setState({buttonWidth:width,buttonHeight:height,buttonPositionX:locationX,buttonPositionY:locationY});}</span>else{<span class="cstat-no" title="statement not covered" >_this.marker.measure(<span class="fstat-no" title="function not covered" >fu</span>nction(x,y,width,height,pageX,pageY){<span class="cstat-no" title="statement not covered" >locationX=pageX;<span class="cstat-no" title="statement not covered" >l</span>ocationY=pageY;<span class="cstat-no" title="statement not covered" >_</span>this.setState({buttonWidth:width,buttonHeight:height,buttonPositionX:locationX,buttonPositionY:locationY});}</span>);}</span>}</span>;_this.onMenuLayout=<span class="fstat-no" title="function not covered" >fu</span>nction(e){var _e$nativeEvent$layout2=<span class="cstat-no" title="statement not covered" >e.nativeEvent.layout,</span>width=<span class="cstat-no" title="statement not covered" >_e$nativeEvent$layout2.width,</span>height=<span class="cstat-no" title="statement not covered" >_e$nativeEvent$layout2.height;<span class="cstat-no" title="statement not covered" ></span>_this.setState({initialWidth:width,initialHeight:height});}</span>;return _this;}(0,_createClass2.default)(Menu,[{key:"componentDidMount",value:function <span class="fstat-no" title="function not covered" >componentDidMount(</span>){var visible=<span class="cstat-no" title="statement not covered" >this.props.visible;<span class="cstat-no" title="statement not covered" ></span>if(visible){<span class="cstat-no" title="statement not covered" >this.toggle();}</span>}</span>},{key:"componentDidUpdate",value:function <span class="fstat-no" title="function not covered" >componentDidUpdate(</span>prevProps){var visible=<span class="cstat-no" title="statement not covered" >this.props.visible;<span class="cstat-no" title="statement not covered" ></span>if(visible!==prevProps.visible){<span class="cstat-no" title="statement not covered" >this.toggle();}</span>}</span>},{key:"toggle",value:function <span class="fstat-no" title="function not covered" >toggle(</span>){var _this2=<span class="cstat-no" title="statement not covered" >this;</span>var _this$state=<span class="cstat-no" title="statement not covered" >this.state,</span>initialHeight=<span class="cstat-no" title="statement not covered" >_this$state.initialHeight,</span>initialWidth=<span class="cstat-no" title="statement not covered" >_this$state.initialWidth,</span>easing=<span class="cstat-no" title="statement not covered" >_this$state.easing,</span>animationDuration=<span class="cstat-no" title="statement not covered" >_this$state.animationDuration,</span>buttonWidth=<span class="cstat-no" title="statement not covered" >_this$state.buttonWidth;<span class="cstat-no" title="statement not covered" ></span>if(!initialHeight||!initialWidth){<span class="cstat-no" title="statement not covered" >setTimeout(<span class="fstat-no" title="function not covered" >fu</span>nction(){<span class="cstat-no" title="statement not covered" >return _this2.toggle();}</span>,100);<span class="cstat-no" title="statement not covered" >r</span>eturn;}</span>v</span>ar sameWidth=<span class="cstat-no" title="statement not covered" >this.props.sameWidth;</span>var widthType=<span class="cstat-no" title="statement not covered" >sameWidth?buttonWidth:initialWidth;</span>var opacity=<span class="cstat-no" title="statement not covered" >1;</span>var height=<span class="cstat-no" title="statement not covered" >initialHeight+24;</span>var width=<span class="cstat-no" title="statement not covered" >widthType+24;<span class="cstat-no" title="statement not covered" ></span>if(!this.props.visible){<span class="cstat-no" title="statement not covered" >opacity=0;<span class="cstat-no" title="statement not covered" >h</span>eight=0;<span class="cstat-no" title="statement not covered" >w</span>idth=0;}<span class="cstat-no" title="statement not covered" ></span>_</span>Animated.default.parallel([_Animated.default.timing(this.state.menuHeight,{toValue:height,duration:animationDuration,easing:easing}),_Animated.default.timing(this.state.menuWidth,{toValue:width,duration:animationDuration,easing:easing}),_Animated.default.timing(this.state.opacity,{toValue:opacity,duration:animationDuration,easing:easing})]).start();}</span>},{key:"render",value:function render(){var _this3=this;var _this$state2=this.state,menuHeight=_this$state2.menuHeight,menuWidth=_this$state2.menuWidth,opacity=_this$state2.opacity,buttonWidth=_this$state2.buttonWidth,buttonPositionY=_this$state2.buttonPositionY,buttonPositionX=_this$state2.buttonPositionX,modalMenuWidth=_this$state2.modalMenuWidth,initialWidth=_this$state2.initialWidth,initialHeight=_this$state2.initialHeight;var _this$props=this.props,button=_this$props.button,children=_this$props.children,menuStyle=_this$props.menuStyle,modalMenuStyle=_this$props.modalMenuStyle,sameWidth=_this$props.sameWidth,visible=_this$props.visible,onBackdropPress=_this$props.onBackdropPress,contentContainerStyle=_this$props.contentContainerStyle,tooltip=_this$props.tooltip,tooltipPosition=_this$props.tooltipPosition,noBackDrop=_this$props.noBackDrop,testID=_this$props.testID,fullWidth=_this$props.fullWidth;var menuContainerStyle={height:menuHeight,width:menuWidth,opacity:opacity,top:buttonPositionY?<span class="branch-0 cbranch-no" title="branch not covered" >buttonPositionY-14:</span>0};<span class="missing-if-branch" title="if path not taken" >I</span>if(modalMenuWidth&gt;0){<span class="cstat-no" title="statement not covered" >if(buttonPositionX&gt;modalMenuWidth/2){<span class="cstat-no" title="statement not covered" >menuContainerStyle.right=modalMenuWidth-buttonPositionX-buttonWidth;}</span>else{<span class="cstat-no" title="statement not covered" >menuContainerStyle.left=buttonPositionX-10;}</span>}<span class="missing-if-branch" title="if path not taken" >I</span></span>if(tooltip){<span class="cstat-no" title="statement not covered" >menuContainerStyle.top=buttonPositionY-initialHeight-12;<span class="cstat-no" title="statement not covered" >m</span>enuContainerStyle.left=buttonPositionX-8+buttonWidth/2-initialWidth/2;<span class="cstat-no" title="statement not covered" >i</span>f(tooltipPosition==='bottom'){<span class="cstat-no" title="statement not covered" >menuContainerStyle.top=buttonPositionY+initialHeight+(_Platform.default.OS==='web'?4:-4);}</span>else <span class="cstat-no" title="statement not covered" >if(tooltipPosition==='right'){<span class="cstat-no" title="statement not covered" >menuContainerStyle.top=buttonPositionY-(_Platform.default.OS==='web'?4:8);<span class="cstat-no" title="statement not covered" >m</span>enuContainerStyle.left=buttonPositionX-8+buttonWidth;}</span>else <span class="cstat-no" title="statement not covered" >if(tooltipPosition==='left'){<span class="cstat-no" title="statement not covered" >menuContainerStyle.top=buttonPositionY-(_Platform.default.OS==='web'?4:8);<span class="cstat-no" title="statement not covered" >m</span>enuContainerStyle.left=buttonPositionX-12-initialWidth;}</span>}<span class="missing-if-branch" title="if path not taken" >I</span></span></span></span>if(Number.isNaN(menuContainerStyle.top)||Number.isNaN(menuContainerStyle.left)){<span class="cstat-no" title="statement not covered" >menuContainerStyle.top=0;<span class="cstat-no" title="statement not covered" >m</span>enuContainerStyle.left=0;}</span>return _react.default.createElement(_View.default,{testID:testID,__self:this,__source:{fileName:_jsxFileName,lineNumber:207,columnNumber:7}},_react.default.createElement(_View.default,{onLayout:this.onButtonLayout,ref:function <span class="fstat-no" title="function not covered" >ref(</span>_ref){<span class="cstat-no" title="statement not covered" >_this3.marker=_ref;}</span>,style:{alignSelf:'flex-start',width:fullWidth?<span class="branch-0 cbranch-no" title="branch not covered" >'100%':</span>'auto'},__self:this,__source:{fileName:_jsxFileName,lineNumber:208,columnNumber:9}},button),_react.default.createElement(_ModalMenu.default,{onLayout:this.onModalMenuLayout,animationType:"none",visible:visible,onBackdropPress:onBackdropPress,noBackDrop:noBackDrop,transparent:true,__self:this,__source:{fileName:_jsxFileName,lineNumber:220,columnNumber:9}},_react.default.createElement(_Animated.default.View,{useNativeDriver:false,style:[_Menu.default.menuContainer,menuContainerStyle,modalMenuStyle],__self:this,__source:{fileName:_jsxFileName,lineNumber:228,columnNumber:11}},_react.default.createElement(_View.default,{style:[_Menu.default.menu,{width:sameWidth?<span class="branch-0 cbranch-no" title="branch not covered" >buttonWidth:</span>'auto'},menuStyle],onLayout:this.onMenuLayout,__self:this,__source:{fileName:_jsxFileName,lineNumber:232,columnNumber:13}},_react.default.createElement(_ScrollView.default,{contentContainerStyle:[{paddingVertical:8},contentContainerStyle],__self:this,__source:{fileName:_jsxFileName,lineNumber:236,columnNumber:15}},children)))));}}]);return Menu;}(_react.Component);Menu.propTypes={button:_propTypes.default.node,children:_propTypes.default.node,visible:_propTypes.default.bool,menuStyle:_propTypes.default.oneOfType([_propTypes.default.object,_propTypes.default.array]),sameWidth:_propTypes.default.bool,onBackdropPress:_propTypes.default.func,modalMenuStyle:_propTypes.default.oneOfType([_propTypes.default.object,_propTypes.default.array]),contentContainerStyle:_propTypes.default.oneOfType([_propTypes.default.object,_propTypes.default.array]),noBackDrop:_propTypes.default.bool,tooltip:_propTypes.default.bool,tooltipPosition:_propTypes.default.string,testID:_propTypes.default.string,fullWidth:_propTypes.default.bool};var _default=Menu;exports.default=_default;</pre></td></tr></table></pre>

                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Sun Jul 11 2021 09:12:08 GMT-0600 (Mountain Daylight Time)
            </div>
        </div>
        <script src="../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../sorter.js"></script>
        <script src="../../../block-navigation.js"></script>
    </body>
</html>
    