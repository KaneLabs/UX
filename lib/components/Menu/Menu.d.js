var _react=require("react");