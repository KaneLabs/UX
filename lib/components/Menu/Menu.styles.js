var _interopRequireDefault=require("@babel/runtime/helpers/interopRequireDefault");Object.defineProperty(exports,"__esModule",{value:true});exports.default=void 0;var _extends2=_interopRequireDefault(require("@babel/runtime/helpers/extends"));var _StyleSheet=_interopRequireDefault(require("react-native-web/dist/exports/StyleSheet"));var _Shadow=_interopRequireDefault(require("../Shadow"));var styles=_StyleSheet.default.create({menuContainer:{top:-10,overflow:'hidden',position:'absolute',zIndex:200},menu:(0,_extends2.default)({borderRadius:6,position:'absolute',left:10,top:10,backgroundColor:'white',minWidth:112,zIndex:200},(0,_Shadow.default)(8))});var _default=styles;exports.default=_default;