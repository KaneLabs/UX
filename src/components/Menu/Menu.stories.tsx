import React from 'react';
import { View } from 'react-native';

import { State, Store } from '@sambego/storybook-state';
import { Menu, MenuItem, Button, IconButton, Heading } from '../..';
import Header from '@kanelabs/ux-storybook/components/Header';
import Container from '@kanelabs/ux-storybook/components/Container';
import { storiesOf } from '@kanelabs/ux-storybook/helpers/storiesOf';

const store = new Store({
  visibleOne: false,
  visibleTwo: false,
  visibleThree: false,
  visibleFour: false,
  visibleFive: false,
});

export default storiesOf('Components|Menu', module)
  .addParameters({ jest: ['Menu'] })
  .add('Simple', () => (
    <Container>
      <Header title="Menu" />
      <Heading style={{ marginBottom: 10 }}>On the left</Heading>
      <State store={store} style={{ flex: 1 }}>
        {(state) => (
          <View style={{ marginBottom: 40 }}>
            <Menu
              menuStyle={{ width: 324 }}
              visible={state.visibleOne}
              onBackdropPress={() => {
                store.set({ visibleOne: false });
              }}
              button={
                <Button
                  text="Show menu"
                  onPress={() => {
                    store.set({ visibleOne: !state.visibleOne });
                  }}
                  type="contained"
                />
              }>
              <MenuItem
                text="Menu Item 1"
                onPress={() => store.set({ visibleOne: false })}
              />
              <MenuItem
                text="Menu Item 2"
                onPress={() => store.set({ visibleOne: false })}
              />
              <MenuItem
                text="Menu Item 3"
                disabled
                onPress={() => store.set({ visibleOne: false })}
              />
              <MenuItem
                text="Menu Item 4"
                onPress={() => store.set({ visibleOne: false })}
              />
            </Menu>
          </View>
        )}
      </State>
      <State store={store} style={{ flex: 1 }}>
        {(state) => (
          <View>
            <Menu
              menuStyle={{ width: 324 }}
              visible={state.visibleTwo}
              onBackdropPress={() => {
                store.set({ visibleTwo: false });
              }}
              button={
                <IconButton
                  name="more-vert"
                  onPress={() => {
                    store.set({ visibleTwo: !state.visibleTwo });
                  }}
                  size={24}
                />
              }>
              <MenuItem
                text="Menu Item 1"
                onPress={() => store.set({ visibleTwo: false })}
              />
              <MenuItem
                text="Menu Item 2"
                onPress={() => store.set({ visibleTwo: false })}
              />
              <MenuItem
                text="Menu Item 3"
                disabled
                onPress={() => store.set({ visibleTwo: false })}
              />
              <MenuItem
                text="Menu Item 4"
                onPress={() => store.set({ visibleTwo: false })}
              />
            </Menu>
          </View>
        )}
      </State>
      <Heading
        style={{ marginBottom: 10, marginTop: 50, alignSelf: 'flex-end' }}>
        On the Right
      </Heading>
      <State store={store} style={{ flex: 1 }}>
        {(state) => (
          <View style={{ alignItems: 'flex-end', marginBottom: 40 }}>
            <Menu
              visible={state.visibleThree}
              onBackdropPress={() => {
                store.set({ visibleThree: false });
              }}
              sameWidth
              button={
                <Button
                  text="Show menu"
                  onPress={() => {
                    store.set({ visibleThree: !state.visibleThree });
                  }}
                  type="contained"
                />
              }>
              <MenuItem
                text="Menu Item 1"
                style={{ paddingHorizontal: '12%' }}
                onPress={() => store.set({ visibleThree: false })}
              />
              <MenuItem
                text="Menu Item 2"
                style={{ paddingHorizontal: '12%' }}
                onPress={() => store.set({ visibleThree: false })}
              />
              <MenuItem
                text="Menu Item 3"
                style={{ paddingHorizontal: '12%' }}
                disabled
                onPress={() => store.set({ visibleThree: false })}
              />
              <MenuItem
                text="Menu Item 4"
                style={{ paddingHorizontal: '12%' }}
                onPress={() => store.set({ visibleThree: false })}
              />
            </Menu>
          </View>
        )}
      </State>
      <State store={store} style={{ flex: 1 }}>
        {(state) => (
          <View style={{ alignItems: 'flex-end' }}>
            <Menu
              visible={state.visibleFour}
              onBackdropPress={() => {
                store.set({ visibleFour: false });
              }}
              button={
                <IconButton
                  name="more-vert"
                  onPress={() => {
                    store.set({ visibleFour: !state.visibleFour });
                  }}
                  size={24}
                />
              }>
              <MenuItem
                text="Menu Item 1"
                onPress={() => store.set({ visibleFour: false })}
              />
              <MenuItem
                text="Menu Item 2"
                onPress={() => store.set({ visibleFour: false })}
              />
              <MenuItem
                text="Menu Item 3"
                disabled
                onPress={() => store.set({ visibleFour: false })}
              />
              <MenuItem
                text="Menu Item 4"
                onPress={() => store.set({ visibleFour: false })}
              />
            </Menu>
          </View>
        )}
      </State>
    </Container>
  ))
  .add('Scrolling', () => (
    <Container>
      <Header title="Menu" />

      <State store={store} style={{ flex: 1 }}>
        {(state) => (
          <View style={{ marginBottom: 40 }}>
            <Menu
              menuStyle={{ width: 324, maxHeight: 200 }}
              visible={state.visibleFive}
              onBackdropPress={() => {
                store.set({ visibleFive: false });
              }}
              button={
                <Button
                  text="Show menu"
                  onPress={() => {
                    store.set({ visibleFive: !state.visibleFive });
                  }}
                  type="contained"
                />
              }>
              <MenuItem
                text="Menu Item 1"
                onPress={() => store.set({ visibleFive: false })}
              />
              <MenuItem
                text="Menu Item 2"
                onPress={() => store.set({ visibleFive: false })}
              />
              <MenuItem
                text="Menu Item 3"
                disabled
                onPress={() => store.set({ visibleFive: false })}
              />
              <MenuItem
                text="Menu Item 4"
                onPress={() => store.set({ visibleFive: false })}
              />
              <MenuItem
                text="Menu Item 5"
                onPress={() => store.set({ visibleFive: false })}
              />
              <MenuItem
                text="Menu Item 6"
                onPress={() => store.set({ visibleFive: false })}
              />
              <MenuItem
                text="Menu Item 7"
                disabled
                onPress={() => store.set({ visibleFive: false })}
              />
              <MenuItem
                text="Menu Item 8"
                onPress={() => store.set({ visibleFive: false })}
              />
            </Menu>
          </View>
        )}
      </State>
    </Container>
  ));
