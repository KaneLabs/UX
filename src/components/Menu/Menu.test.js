import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Menu from './Menu';

const renderer = new ShallowRenderer();

test('Menu Renders', () => {
  const tree = renderer.render(<Menu />);
  expect(tree).toMatchSnapshot();
});
