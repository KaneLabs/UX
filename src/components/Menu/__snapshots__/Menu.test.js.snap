// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Renders 1`] = `
<View>
  <View
    onLayout={[Function]}
    style={
      Object {
        "alignSelf": "flex-start",
        "width": "auto",
      }
    }
  />
  <Modal
    animationType="none"
    onLayout={[Function]}
    transparent={true}
    visible={false}
  >
    <ForwardRef(AnimatedComponentWrapper)
      style={
        Array [
          Object {
            "overflow": "hidden",
            "position": "absolute",
            "top": -10,
            "zIndex": 200,
          },
          Object {
            "height": 0,
            "opacity": 0,
            "top": 0,
            "width": 0,
          },
          undefined,
        ]
      }
      useNativeDriver={false}
    >
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "borderRadius": 6,
              "left": 10,
              "minWidth": 112,
              "position": "absolute",
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 4,
                "width": 0,
              },
              "shadowOpacity": 0.3,
              "shadowRadius": 4.65,
              "top": 10,
              "zIndex": 200,
            },
            Object {
              "width": "auto",
            },
            undefined,
          ]
        }
      >
        <ScrollView
          contentContainerStyle={
            Array [
              Object {
                "paddingVertical": 8,
              },
              undefined,
            ]
          }
        />
      </View>
    </ForwardRef(AnimatedComponentWrapper)>
  </Modal>
</View>
`;
